modules/files.nix

```
@@ -14,6 +14,19 @@ let
     inherit homeDirectory lib pkgs;
   }).fileType;
 
+  sourceStorePath = file:
+    let
+      srcText = toString file.source;
+      base = baseNameOf srcText;
+    in
+      if builtins.hasContext srcText
+      then file.source
+      else
+        builtins.path {
+          path = file.source;
+          name = if hasPrefix "." base then substring 1 (-1) base else base;
```

If I understand correctly, this would still fail on a file like `..blah`. 
```suggestion
          name = if hasPrefix "." base then "file_" + base else base;
```

---

That's even better indeed !